### ISSUE TITLE
runCatching inside flow.map throw an error : org.jetbrains.kotlin.codegen.CompilationException: Couldn't transform method node

### ISSUE THREAD

**Issue by scorsi** [Kotlin/kotlinx.coroutines#2097]
I don't know if I have to post the issue here on in the main Kotlin repository.

I have to following code which can't be compiled... I reduced to the maximum:
```kotlin
runBlocking {
    someList
        .asFlow()
        .map {
            runCatching {
                json.writeValueAsString(it)
            }.getOrNull()
        }
        .collect()
}
```
But this one can be compiled:
```kotlin
runBlocking {
    someList
        .asFlow()
        .map {
            json.writeValueAsString(it)
        }
        .collect()
}
```

`runCatching()` inside the `flow.map()` can't be compiled.. Do you have any idea ?

I'm running with:
 - plugin `org.jetbrains.kotlin.jvm`: 1.3.72
 - dep `org.jetbrains.kotlin:kotlin-stdlib-jdk8`: 1.3.72
 - dep `org.jetbrains.kotlinx:kotlinx-coroutines-core`: 1.3.7
 - os : MacOS Catalina 10.15.4
 - JVM/JDK : 13.0.1

To the the stacktrace, it's [here](https://gist.github.com/scorsi/61f54f51dfbebfa5707aae36aee8ae06) (very long one with weird Kotlin-compiler AST printed).

**Comment by fvasco** [comment]
See https://youtrack.jetbrains.com/issue/KT-34708

**Comment by scorsi** [comment]
Oh, I didn't thought about looking at YouTrack sorry ^^
Did we close that issue now or maybe we wait until the issue on YouTrack is closed ?

**Comment by fvasco** [comment]
`KT-34708` has been already fixed and released in Kotlin 1.4-M1

**Comment by scorsi** [comment]
Oh ok, I don’t really want to use the pre-release of Kotlin even if some new features are really cool (like SAM interface/constructor).
Thank you !
